src/__tests__/components/frontend-engine/frontend-engine.spec.tsx

```diff
@@ -0,0 +1,188 @@
+import { fireEvent, render, screen, waitFor } from "@testing-library/react";
+import { useRef } from "react";
+import { FrontendEngine } from "src/components";
+import { IFrontendEngineData, IFrontendEngineProps, IFrontendEngineRef } from "../../../components/types";
```

Comment:
when do we decide to use absolute or relative imports?

Reply:
seems like there is no hard-and-fast rule so I decided to switch everything to relative imports because it appears to be the more standard approach. Along the same line, I updated the vscode `settings.json` to use relative imports by default to standardise it.

and just a wild thought, maybe we can have our own custom linter in the future to catch such issues.